tests/components/hue/test_sensor_base.py

```diff
@@ -477,13 +478,29 @@ async def test_hue_events(hass, mock_bridge):
 
     assert len(mock_bridge.mock_requests) == 3
     assert len(hass.states.async_all()) == 7
-    assert len(mock_listener.mock_calls) == 2
-    assert mock_listener.mock_calls[1][1][0].data == {
+    assert len(events) == 2
+    assert events[-1].data == {
         "id": "hue_dimmer_switch_1",
         "unique_id": "00:17:88:01:10:3e:3a:dc-02-fc00",
         "event": 3002,
-        "last_updated": "2019-12-28T22:58:01",
+        "last_updated": "2019-12-28T22:58:03",
+    }
+
+    # Fire old event, it should be ignored
+    new_sensor_response = dict(new_sensor_response)
+    new_sensor_response["8"]["state"] = {
+        "buttonevent": 18,
+        "lastupdated": "2019-12-28T22:58:02",
     }
+    mock_bridge.mock_sensor_responses.append(new_sensor_response)
+
+    # Force updates to run again
+    await mock_bridge.sensor_manager.coordinator.async_refresh()
```

Comment:
Normally we try to avoid accessing integration details in test for integrations. We could move time forward instead.

Reply:
https://github.com/home-assistant/core/pull/50783